tests/test_interpolate.py

```diff
@@ -351,3 +352,623 @@ def test_duplicate_idxs_period_pairs():
+    lhs_angle_k = lhs.angle.params[0][0]
+    lhs_angle_a0 = lhs.angle.params[0][1]
+
+    rhs_bond_k = rhs.bond.params[0][0]
+    rhs_bond_b0 = rhs.bond.params[0][1]
+    rhs_angle_k = rhs.angle.params[0][0]
+    rhs_angle_a0 = rhs.angle.params[0][1]
+
+    # no chiral conversion happens, so we should be using the full schedule for bonds and angles
+    atol_k = 1e-3
+    atol_b0 = 1e-5
+    atol_a0 = 1e-4
+    for lamb in np.linspace(0.05, 0.95, 12):
+        itm = st.setup_intermediate_state(lamb)
+        itm_bond_k = itm.bond.params[0][0]
+        itm_bond_b0 = itm.bond.params[0][1]
+        itm_angle_k = itm.angle.params[0][0]
+        itm_angle_a0 = itm.angle.params[0][1]
+
+        assert abs(itm_bond_k - lhs_bond_k) > atol_k
+        assert abs(itm_bond_b0 - lhs_bond_b0) > atol_b0
+        assert abs(itm_angle_k - lhs_angle_k) > atol_k
+        assert abs(itm_angle_a0 - lhs_angle_a0) > atol_a0
+
+        assert abs(itm_bond_k - rhs_bond_k) > atol_k
+        assert abs(itm_bond_b0 - rhs_bond_b0) > atol_b0
+        assert abs(itm_angle_k - rhs_angle_k) > atol_k
+        assert abs(itm_angle_a0 - rhs_angle_a0) > atol_a0
+
+
+@pytest.mark.parametrize(
+    "mol_a, mol_b, core",
+    [
+        (get_aldehyde(), get_non_inv_nitrogen(), np.array([[0, 0], [1, 1], [2, 2], [3, 3]], dtype=np.int32)),
+        (get_inv_nitrogen(), get_non_inv_nitrogen(), np.array([[0, 0], [1, 1], [2, 2], [3, 3]], dtype=np.int32)),
+    ],
+)
+def test_core_chiral_interpolation_sp2_to_invertible_sp3_fwd(mol_a, mol_b, core):
+    ff = Forcefield.load_default()
+    st = SingleTopology(mol_a, mol_b, core, ff)
+
+    lhs = st.setup_intermediate_state(0.0)
+    rhs = st.setup_intermediate_state(1.0)
+
+    assert len(lhs.chiral_atom.potential.idxs) == 1
+    assert len(rhs.chiral_atom.potential.idxs) == 1
+    np.testing.assert_array_equal(lhs.chiral_atom.potential.idxs, rhs.chiral_atom.potential.idxs)
+
+    assert lhs.chiral_atom.params[0] == 0
+    assert rhs.chiral_atom.params[0] == DEFAULT_CHIRAL_ATOM_RESTRAINT_K
+
+    lhs_bond_k = lhs.bond.params[0][0]
+    lhs_bond_b0 = lhs.bond.params[0][1]
+    lhs_angle_k = lhs.angle.params[0][0]
+    lhs_angle_a0 = lhs.angle.params[0][1]
+    lhs_chiral_k = lhs.chiral_atom.params[0]
+
+    rhs_bond_k = rhs.bond.params[0][0]
+    rhs_bond_b0 = rhs.bond.params[0][1]
+    rhs_angle_k = rhs.angle.params[0][0]
+    rhs_angle_a0 = rhs.angle.params[0][1]
+    rhs_chiral_k = rhs.chiral_atom.params[0]
+
+    # # no chiral conversion happens, so we should be using the full schedule for bonds and angles
+    atol_k = 1e-3
+    atol_b0 = 1e-5
+    atol_a0 = 1e-4
+    for lamb in np.linspace(0.05, 0.95, 12):
+        itm = st.setup_intermediate_state(lamb)
+        itm_bond_k = itm.bond.params[0][0]
+        itm_bond_b0 = itm.bond.params[0][1]
+        itm_angle_k = itm.angle.params[0][0]
+        itm_angle_a0 = itm.angle.params[0][1]
+        itm_chiral_k = itm.chiral_atom.params[0]
+
+        # bonds are lambda independent
```

Comment:
Should this say "bonds are interpolated over the full range of lambda"? (They don't seem lambda-independent, comparing with assertions immediately below)